database/rotator.go

```diff
@@ -239,29 +264,25 @@ func rotatePartitions(db *dbr.Connection, query string, d, p int) {
 	t := time.Now().Add(time.Hour * time.Duration(24*d))
 	oldName := "pnr0000"
 	newName := "pnr0"
+	OriQuery := query
 
 	startTime := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location())
 	oldStart := "StartTime"
-	newStart := startTime.Add(time.Minute * time.Duration(0)).Format("2006-01-02 15:04:05-07:00")
 
 	endTime := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location())
 	oldEnd := "EndTime"
-	newEnd := endTime.Add(time.Minute * time.Duration(p)).Format("2006-01-02 15:04:05-07:00")
 
-	for i := 0; i < 1440/p; i++ {
-		if i > 0 {
-			newName = "pnr" + strconv.Itoa(i)
-			newStart = startTime.Add(time.Minute * time.Duration(i*p)).Format("2006-01-02 15:04:05-07:00")
-			newEnd = endTime.Add(time.Minute * time.Duration(i*p+p)).Format("2006-01-02 15:04:05-07:00")
-		}
+	for i := 1; i < 1440/p; i++ {
+		query := OriQuery
```

Comment:
Re-use the original query (with "StartTime" and "EndTime") every loop to replace the time

Reply:
For loop, might have to start with 0 instead of 1, was not thinking about the postgres here.
Because for mysql the first partition is added already thats why i used 1 to skip the first partition.